nixops/backends/ec2.py

```diff
@@ -458,8 +461,11 @@ def restore(self, defn, backup_id, devices=[]):
         for d in devices:
             self.log(" - {0}".format(d))
 
-        for k, v in self.block_device_mapping.items():
-            if devices == [] or _sd_to_xvd(k) in devices:
+        # because name of the nvme device depends on the order it attached to maching
+        sorted_block_device_mapping = sorted(self.block_device_mapping.items())
```

Comment:
this code and comment is used in two places. What about moving it closer to initialization? So walk order is unified and deterministic.

Reply:
also there is a typo in comment `maching` -> `machine`. But even with typo fixed I don't understand comment. Maybe rephrase: "Volumes should be attached in lexicographic order (ordered by device name). This will preserve nvme devices' names"